Tribler/Core/CacheDB/SqliteCacheDBHandler.py

```diff
@@ -979,15 +979,20 @@ def searchNames(self, kws, local=True, keys=None, doSort=True):
             result.extend(channel)
 
             if doSort and result[num_seeders_index] <= 0:
-                dont_sort_list.append(result)
-                results.pop(i)
-
+                dont_sort_list.append((index, result))
 
         if doSort:
```

Comment:
Do you mean that the if on line 984 should be moved inside the if on line 981? Note that the if on line 981 is in a for-loop.